### PR TITLE
Add some test coverage

### PR DESCRIPTION
**Pull request by cmars**
Consolidated test table initialization, added tests for serialization, etc. This gets coverage up to 90%.

### CODE REVIEW

**Comment by vadimg** [comment]
Wow, thank you for the tests! But, it seems one of the XML tests fails in go 1.1. Can you investigate why?

If the reason is a go 1.1 bug, I'm perfectly happy to remove go 1.1 support from this library.

**Comment by cmars** [comment]
The `encoding` package interfaces `TextMarshaler` and `TextUnmarshaler` were added in Go 1.2. Go 1.1 does not use `MarshalText`/`UnmarshalText` for XML serialization.

**Comment by vadimg** [comment]
Ok, in that case, can you remove go 1.1 from .travis.yml?

File: decimal_test.go [hunk 1]
```diff
@@ -431,6 +528,23 @@ func TestDecimal_Abs2(t *testing.T) {
 	}
 }
 
+func TestDecimal_Equal(t *testing.T) {
+	a := New(1234, 3)
+	b := New(1234, 3)
+
+	if !a.Equals(b) {
+		t.Errorf("Error")
```
Comment (vadimg) [inline decimal_test.go]: your error messages should be more descriptive

File: decimal_test.go [hunk 2]
```diff
@@ -431,6 +528,23 @@ func TestDecimal_Abs2(t *testing.T) {
...
+	}
+}
+
+func TestDecimal_ScalesNotEqual(t *testing.T) {
+	a := New(1234, 2)
+	b := New(1234, 3)
+	if a.Equals(b) {
+		t.Errorf("Error")
```
Comment (vadimg) [inline decimal_test.go]: here too, you should say what did not equal

File: decimal_test.go [hunk 3]
```diff
@@ -152,6 +149,91 @@ func TestNewFromFloatWithExponent(t *testing.T) {
...
+			t.Errorf("error marshaling %+v: %v", doc, err)
+		} else if string(out) != docStr {
+			t.Errorf("expected %s, got %s", docStr, string(out))
+		}
+	}
+}
+
+func TestBadJSON(t *testing.T) {
+	for _, testCase := range []string{
+		"]o_o[", "{", `{"amount":""`,
```
Comment (vadimg) [inline decimal_test.go]: could you please put each of these testcases on a separate line? It's hard to follow otherwise.

File: decimal_test.go [hunk 4]
```diff
@@ -152,6 +149,91 @@ func TestNewFromFloatWithExponent(t *testing.T) {
...
+			t.Errorf("error marshaling %+v: %v", doc, err)
+		} else if string(out) != docStr {
+			t.Errorf("expected %s, got %s", docStr, string(out))
+		}
+	}
+}
+
+func TestBadXML(t *testing.T) {
+	for _, testCase := range []string{
+		"o_o", "<abc", "<account><amount>7",
```
Comment (vadimg) [inline decimal_test.go]: could you please put each of these testcases on a separate line? It's hard to follow otherwise.

File: decimal_test.go [hunk 5]
```diff
@@ -402,6 +484,21 @@ func TestDecimal_Overflow(t *testing.T) {
 	}
 }
 
+func TestIntPart(t *testing.T) {
+	for _, testCase := range []struct {
+		Dec     string
+		IntPart int64
+	}{{"0.01", 0}, {"12.1", 12}, {"9999.999", 9999}, {"-32768.01234", -32768}} {
```
Comment (vadimg) [inline decimal_test.go]: could you please put each of these testcases on a separate line? It's hard to follow otherwise.

**Comment by cmars** [comment]
PTAL
